.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,44 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Next.js Shopify Starter
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - './**'
+  pull_request:
+    branches: [main]
+    paths:
+      - './**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: './package.json'
+          cache: ''
+
+      - name: Install dependencies
+        run: npm i
+        working-directory: ./
+
+      - name: Lint files
+        run: npm run lint && npm run format
+        working-directory: ./
+
+      - name: Build Project
+        run: npm run build
+        working-directory: ./
+
+      - name: Test Project
+        run: npm run test
+        working-directory: ./
```

Comment:
I think we could remove this step for now. We don't have tests and probably this command will always fails